From 6ce4601cef72a630f7b53c98c5b0683ccc509986 Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Sat, 20 Feb 2016 14:18:26 +0000
Subject: [PATCH] filemanager/panel.c: cleanup -Wcast-qual warning

In file included from ../../lib/widget.h:18:0,
                 from panel.c:53:
panel.c: In function 'panel_lines':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
panel.c:409:13: note: in expansion of macro 'WIDGET'
     return (WIDGET (p)->lines - 3 - (panels_options.show_mini_info ? 2 : 0));
             ^
panel.c: In function 'display_total_marked_size':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
panel.c:1070:17: note: in expansion of macro 'WIDGET'
     Widget *w = WIDGET (panel);
                 ^
panel.c:1094:11: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
     buf = (char *) str_trunc (buf, cols - 4);
           ^
In file included from ../../lib/widget.h:18:0,
                 from panel.c:53:
panel.c: In function 'mini_info_separator':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
panel.c:1116:21: note: in expansion of macro 'WIDGET'
         Widget *w = WIDGET (panel);
                     ^
panel.c: In function 'show_free_space':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
panel.c:1159:21: note: in expansion of macro 'WIDGET'
         Widget *w = WIDGET (panel);
                     ^
panel.c: In function 'show_dir':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
panel.c:1260:17: note: in expansion of macro 'WIDGET'
     Widget *w = WIDGET (panel);
                 ^
panel.c: In function 'panel_paint_sort_info':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
../../lib/widget/widget-common.h:15:44: note: in expansion of macro 'WIDGET'
 #define widget_move(w, _y, _x) tty_gotoyx (WIDGET(w)->y + (_y), WIDGET(w)->x + (_x))
                                            ^
panel.c:1524:9: note: in expansion of macro 'widget_move'
         widget_move (panel, 1, 1);
         ^
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
../../lib/widget/widget-common.h:15:65: note: in expansion of macro 'WIDGET'
 #define widget_move(w, _y, _x) tty_gotoyx (WIDGET(w)->y + (_y), WIDGET(w)->x + (_x))
                                                                 ^
panel.c:1524:9: note: in expansion of macro 'widget_move'
         widget_move (panel, 1, 1);
         ^
panel.c: In function 'panel_print_header':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                    ^
panel.c:1558:17: note: in expansion of macro 'WIDGET'
     Widget *w = WIDGET (panel);
                 ^
panel.c: In function 'panel_get_sortable_fields':
panel.c:4879:12: warning: to be safe all intermediate pointers in cast from 'char **' to 'const char **' must be 'const' qualified [-Wcast-qual]
     return (const char **) ret;
            ^
panel.c: In function 'panel_get_user_possible_fields':
panel.c:4967:12: warning: to be safe all intermediate pointers in cast from 'char **' to 'const char **' must be 'const' qualified [-Wcast-qual]
     return (const char **) ret;
            ^

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 src/filemanager/panel.c | 25 +++++++++++++------------
 src/filemanager/panel.h |  4 ++--
 2 files changed, 15 insertions(+), 14 deletions(-)

diff --git a/src/filemanager/panel.c b/src/filemanager/panel.c
index 1fce969..1509b73 100644
--- a/src/filemanager/panel.c
+++ b/src/filemanager/panel.c
@@ -406,7 +406,7 @@ static int
 panel_lines (const WPanel * p)
 {
     /* 3 lines are: top frame, column header, botton frame */
-    return (WIDGET (p)->lines - 3 - (panels_options.show_mini_info ? 2 : 0));
+    return (CWIDGET (p)->lines - 3 - (panels_options.show_mini_info ? 2 : 0));
 }
 
 /* --------------------------------------------------------------------------------------------- */
@@ -1067,9 +1067,10 @@ paint_dir (WPanel * panel)
 static void
 display_total_marked_size (const WPanel * panel, int y, int x, gboolean size_only)
 {
-    Widget *w = WIDGET (panel);
+    const Widget *w = CWIDGET (panel);
 
-    char buffer[BUF_SMALL], b_bytes[BUF_SMALL], *buf;
+    char buffer[BUF_SMALL], b_bytes[BUF_SMALL];
+    const char *buf;
     int cols;
 
     if (panel->marked <= 0)
@@ -1091,7 +1092,7 @@ display_total_marked_size (const WPanel * panel, int y, int x, gboolean size_onl
                     b_bytes, panel->marked);
 
     /* don't forget spaces around buffer content */
-    buf = (char *) str_trunc (buf, cols - 4);
+    buf = str_trunc (buf, cols - 4);
 
     if (x < 0)
         /* center in panel */
@@ -1113,7 +1114,7 @@ mini_info_separator (const WPanel * panel)
 {
     if (panels_options.show_mini_info)
     {
-        Widget *w = WIDGET (panel);
+        const Widget *w = CWIDGET (panel);
         int y;
 
         y = panel_lines (panel) + 2;
@@ -1156,7 +1157,7 @@ show_free_space (const WPanel * panel)
 
     if (myfs_stats.avail != 0 || myfs_stats.total != 0)
     {
-        Widget *w = WIDGET (panel);
+        const Widget *w = CWIDGET (panel);
         char buffer1[6], buffer2[6], tmp[BUF_SMALL];
 
         size_trunc_len (buffer1, sizeof (buffer1) - 1, myfs_stats.avail, 1,
@@ -1257,7 +1258,7 @@ panel_get_encoding_info_str (const WPanel * panel)
 static void
 show_dir (const WPanel * panel)
 {
-    Widget *w = WIDGET (panel);
+    const Widget *w = CWIDGET (panel);
 
     gchar *tmp;
 
@@ -1555,7 +1556,7 @@ panel_get_title_without_hotkey (const char *title)
 static void
 panel_print_header (const WPanel * panel)
 {
-    Widget *w = WIDGET (panel);
+    const Widget *w = CWIDGET (panel);
 
     int y, x;
     int i;
@@ -4855,7 +4856,7 @@ panel_get_num_of_sortable_fields (void)
 
 /* --------------------------------------------------------------------------------------------- */
 
-const char **
+char **
 panel_get_sortable_fields (gsize * array_size)
 {
     char **ret;
@@ -4876,7 +4877,7 @@ panel_get_sortable_fields (gsize * array_size)
         if (panel_fields[i].is_user_choice)
             ret[lc_index++] = g_strdup (_(panel_fields[i].title_hotkey));
 
-    return (const char **) ret;
+    return ret;
 }
 
 /* --------------------------------------------------------------------------------------------- */
@@ -4943,7 +4944,7 @@ panel_get_num_of_user_possible_fields (void)
 
 /* --------------------------------------------------------------------------------------------- */
 
-const char **
+char **
 panel_get_user_possible_fields (gsize * array_size)
 {
     char **ret;
@@ -4964,7 +4965,7 @@ panel_get_user_possible_fields (gsize * array_size)
         if (panel_fields[i].use_in_user_format)
             ret[lc_index++] = g_strdup (_(panel_fields[i].title_hotkey));
 
-    return (const char **) ret;
+    return ret;
 }
 
 /* --------------------------------------------------------------------------------------------- */
diff --git a/src/filemanager/panel.h b/src/filemanager/panel.h
index 4f3cd0d..d3f792e 100644
--- a/src/filemanager/panel.h
+++ b/src/filemanager/panel.h
@@ -180,12 +180,12 @@ void do_file_mark (WPanel * panel, int idx, int val);
 gboolean do_panel_cd (WPanel * panel, const vfs_path_t * new_dir_vpath, enum cd_enum cd_type);
 
 gsize panel_get_num_of_sortable_fields (void);
-const char **panel_get_sortable_fields (gsize *);
+char **panel_get_sortable_fields (gsize *);
 const panel_field_t *panel_get_field_by_id (const char *);
 const panel_field_t *panel_get_field_by_title (const char *);
 const panel_field_t *panel_get_field_by_title_hotkey (const char *);
 gsize panel_get_num_of_user_possible_fields (void);
-const char **panel_get_user_possible_fields (gsize *);
+char **panel_get_user_possible_fields (gsize *);
 void panel_set_cwd (WPanel * panel, const vfs_path_t * vpath);
 void panel_set_lwd (WPanel * panel, const vfs_path_t * vpath);
 
